core/src/main/java/uk/gov/gchq/koryphe/iterable/ChainedIterable.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2022 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.koryphe.iterable;
+
+import uk.gov.gchq.koryphe.util.CloseableUtil;
+
+import java.io.Closeable;
+import java.util.Iterator;
+
+/**
+ * @param <T> the type of items in the iterator
+ */
+public class ChainedIterable<T> implements Closeable, Iterable<T> {
+    private final Iterable<? extends Iterable<? extends T>> iterables;
+
+    public ChainedIterable(final Iterable<? extends Iterable<? extends T>> iterables) {
```

Comment:
The Gaffer version of `ChainedIterable` has another constructor. If you are going to get rid of the Gaffer equivalent and point at the Koryphe version then you may want to add it

Reply:
Good shout, I made them public but forgot to look at Gaffer and move across any constructors or tests